classes/submission/reviewAssignment/ReviewAssignmentDAO.inc.php

```diff
@@ -376,15 +375,21 @@ function insertObject($reviewAssignment) {
 		);
 
 		$reviewAssignment->setId($this->getInsertId());
-		return $reviewAssignment->getId();
+
+		// Update review stage status whenever a review assignment is changed
+		if ($result) {
+			$this->updateReviewRoundStatus($reviewAssignment);
+		}
+
+		return $reviewAssignment;
 	}
 
 	/**
 	 * Update an existing review assignment.
 	 * @param $reviewAssignment object
 	 */
```

Comment:
As above, I don't think this'll return what you expect it to -- but regardless, needs a return self-doc if there will be a return value.